src/commands/positive.ts

```diff
@@ -250,20 +256,27 @@ export default class Positive extends Command {
           scheme: scheme.scheme,
         };
       });
-    // Wrap if statements in loop to iterate over securities array to check for apiKey or bearer_token(s)
+
     let apiKey = flags.apiKey;
     let token = flags.bearerToken;
+    const noPrompt = flags.noPrompt;
     for (const security of securities) {
       if (security.type === OASSecurityType.APIKEY) {
         const apiSecurityName = security.key;
-        const apiKeyValue =
-          apiKey ??
-          // eslint-disable-next-line no-await-in-loop
-          (await cli.prompt('Please provide your API Key', {
-            type: 'mask',
-          }));
+        let apiKeyValue;
+        if (!noPrompt) {
+          apiKeyValue =
+            apiKey ??
+            // eslint-disable-next-line no-await-in-loop
+            (await cli.prompt('Please provide your API Key', {
+              type: 'mask',
+            }));
+        }
+        // if apiKey is undefined, throw an error and provide a message to the user
+        apiKeyValue = apiKey;
         if (!apiKey) {
           apiKey = apiKeyValue;
+          this.log('API key is undefined or empty.');
```

Comment:
I think we should consider throwing an error here and using oclif's error handling https://oclif.io/docs/error_handling

Reply:
This should possibly be done as a separate ticket though... I think refactoring will make this easier

Reply:
Started a ticket here: https://vajira.max.gov/browse/API-14957

Reply:
Started a ticket here: https://vajira.max.gov/browse/API-14957